src/vscode-bicep/src/extension.ts

```diff
@@ -70,11 +84,26 @@ export async function activate(
       new BicepVisualizerViewManager(extension.extensionUri, languageClient)
     );
 
+    const outputChannelManager = new OutputChannelManager(
+      "Bicep Operations",
+      "bicep"
+    );
+
+    // AzLocationTreeItem and AzResourceGroupTreeItem extend AzureAccountTreeItemBase which implements Disposable.
+    // Registering them here:
+    extension.register(new AzLocationTreeItem());
+    extension.register(new AzResourceGroupTreeItem(outputChannelManager));
```

Comment:
This won't dispose the tree items created in the `TreeManager`. The proper way to register them is to let `TreeManager` implement `Disposable` and call `register` in the constructor:
```typescript
import { Disposable } from "../utils/disposable";

export class TreeManager extends Disposable {
  ...

  constructor(private readonly outputChannelManager: OutputChannelManager) {
    const azLocationTreeItem = this.register(new AzLocationTreeItem());
    this._azLocationTree = new AzExtTreeDataProvider(azLocationTreeItem, "");

    const azManagementGroupTreeItem = new AzManagementGroupTreeItem();
    this._azManagementGroupTreeItem = new AzExtTreeDataProvider(
      azManagementGroupTreeItem,
      ""
    );

    const azResourceGroupTreeItem = this.register(
      new AzResourceGroupTreeItem(this.outputChannelManager)
    );
    this._azResourceGroupTreeItem = new AzExtTreeDataProvider(
      azResourceGroupTreeItem,
      ""
    );
  }
  ...
}

```

In `extension.ts`:

```typescript

const treeManager = extension.register(new TreeManager(outputChannelManager));

await extension
      .register(new CommandManager(context))
      .registerCommands(
        new BuildCommand(languageClient, outputChannelManager),
        new DeployCommand(
          languageClient,
          outputChannelManager,
          treeManager
        ),
        new InsertResourceCommand(languageClient),
        new ShowVisualizerCommand(viewManager),
        new ShowVisualizerToSideCommand(viewManager),
        new ShowSourceCommand(viewManager)
      );
  });
```

Reply:
@shenglol , thanks!

Fixed it here:
https://github.com/Azure/bicep/commit/7fc8b0cea94a7af8878ede1d8828c180a1664a2a